301-custom-images-at-scale/azuredeploy.json

```diff
@@ -0,0 +1,355 @@
+            ],
+            "metadata": {
+                "description": "Size of the VM used to transfer the VM image to various storage acounts."
+            }
+        },
+        "computeVMSize": {
+            "type": "string",
+            "defaultValue": "Standard_A1",
+            "allowedValues": [
+                "Standard_A1",
+                "Standard_A2",
+                "Standard_A3",
+                "Standard_A4",
+                "Standard_A5",
+                "Standard_A6",
+                "Standard_A7",
+                "Standard_A8",
+                "Standard_A9",
+                "Standard_A10",
+                "Standard_A11",
+                "Standard_D1",
+                "Standard_D2",
+                "Standard_D3",
+                "Standard_D3_v2",
+                "Standard_D4",
+		"Standard_D4_v2",
+		"Standard_D5_v2",
+                "Standard_D11",
+                "Standard_D12",
+		"Standard_D12_v2",
+                "Standard_D13",
+		"Standard_D13_v2",
+                "Standard_D14",
+		"Standard_D14_v2",
+		"Standard_DS3",
+		"Standard_DS4",
+		"Standard_DS12",
+		"Standard_DS13",
+		"Standard_DS14",
+		"Standard_G2",
+		"Standard_G3",
+		"Standard_G4",
+		"Standard_G5",
+		"Standard_GS2",
+		"Standard_GS3",
+		"Standard_GS4",
+		"Standard_GS5"
+            ],
+            "metadata": {
+                "description": "Size of the VMs to be used for actual computation."
+            }
+        },
+        "computeOSType": {
+            "type": "string",
+            "defaultValue": "Linux",
+            "allowedValues": [
+                "Linux",
+                "Windows"
+            ],
+            "metadata": {
+                "description": "Compute OS Type"
+            }
+        },
+        "deploymentType": {
+            "type": "string",
+            "defaultValue": "VMSS",
+            "allowedValues": [
+                "VMSS",
+                "Single",
+                "SingleAV"
+            ],
+            "metadata": {
+                "description": "This determines whether the VMs will be deployed using scale sets, as individual VMs, or individual VMs in an availability set (maximum 100 for the last option)."
+            }
+        },
+        "numberOfSAs": {
+            "type": "int",
+            "metadata": {
+                "description": "Number of Storage Accounts to upload the custom image to."
+            }
+        },
+        "instanceCountPerSA": {
+            "type": "int",
+            "metadata": {
+                "description": "Number of VMs per Storage Account."
+            }
+        },
+        "imageLocation": {
+            "type": "string",
+            "metadata": {
+                "description": "Location of the base custom image."
+            }
+        },
+        "imageKey": {
+            "type": "string",
+            "metadata": {
+                "description": "Storage Account key for accessing the base custom image."
```

Comment:
If this is just the storage account key, then this can probably be gotten through listKeys function instead of asking the user